### PR TITLE
Cassandra 3.x

### PR DESCRIPTION
**Pull request by magnusart**
Hi

Here is the updated SSL functionality discussed in https://github.com/krasserm/akka-persistence-cassandra/pull/150. I've used cassandra-3.x branch but it looks like it was removed so I hope 3.x-r-0.6 is the correct target? It does look a bit funky that the pull request mentions five commits, I have to admit.

The SSL problem turned out to be expired certificates (DOH!) and a corrupt trust store. 

I also realized that the cassandra setup allows for much more complexity with separate key-/truststores for server to server and server to client communication, so I setup that scenario to make sure config works all the way. 

Recreating the key store setup took longer time than anticipated, so I documented the steps for future reference.

Ran the tests locally and they all passed (including the new SSL-tests)

### CODE REVIEW

**Comment by magnusart** [comment]
SSL-tests succeed. Failing test is an unrelated timeout issue.

File: build.sbt [hunk 1]
```diff
@@ -51,14 +51,14 @@ parallelExecution in Test := false
...
-  "com.datastax.cassandra"  % "cassandra-driver-core"               % "3.0.0-alpha5",
+  "com.datastax.cassandra"  % "cassandra-driver-core"               % "3.0.0-rc1",
   "com.typesafe.akka"      %% "akka-persistence"                    % AkkaVersion,
   "com.typesafe.akka"      %% "akka-persistence-query-experimental" % AkkaVersion,
   "com.typesafe.akka"      %% "akka-persistence-tck"                % AkkaVersion      % "test",
   "org.scalatest"          %% "scalatest"                           % "2.1.4"      % "test",
   "com.typesafe.akka"      %% "akka-stream-testkit-experimental"    % "1.0"        % "test",
   // cassandra-all for testkit.CassandraLauncher, app should define it as test dependency if needed
-  "org.apache.cassandra"    % "cassandra-all"                       % "3.0.0"      % "optional"
+  "org.apache.cassandra"    % "cassandra-all"                       % "3.0.1"      % "optional"
```
Comment (patriknw) [inline build.sbt]: looks like there is a 3.0.2 now

**Comment by patriknw** [comment]
@magnusart what is the target branch here? cassandra-3.x-r-0.6?
please use master, that is the new cassandra-3.x dev branch

**Comment by magnusart** [comment]
@patriknw OK. Didn't know that. I think I have to do a new pull request in that case. I'll update the driver version as well.

File: build.sbt (hunk 1 above)
Comment (magnusart) [inline build.sbt]: Actually, I cannot find any newer driver published yet (at the time of writing).
http://mvnrepository.com/artifact/com.datastax.cassandra/cassandra-driver-core

**Pull request closed without merging**